src/llm-eval/README.md

```diff
@@ -0,0 +1,297 @@
+
+
+## Generate a collection of QAs
+
+The generator can also generate a collection of QAs. The following code shows how to generate 3 sample QAs and it converts the generated collection to JSON.
+
+```csharp
+// generate a collection of QAs using llms
+var llmGenQAs = await QALLMGenerator.GenerateQACollection(kernelGenData, 3);
+
+// convert llmGenQAs to json
+var json = JsonSerializer.Serialize(llmGenQAs, new JsonSerializerOptions
+{
+    WriteIndented = true
+});
+```
+
+This is a sample for the generated json questions:
+
+```json
+{
+    "Question": "What were the primary causes of World War I?",
+    "Answer": "The primary causes of World War I were militarism, alliances, imperialism, and nationalism.",
+    "Topic": "History"
+},
+{
+    "Question": "What is the formula to calculate the area of a circle?",
+    "Answer": "The formula to calculate the area of a circle is A = \u03C0r^2, where A is the area and r is the radius of the circle.",
+    "Topic": "Math"
+},
+{
+    "Question": "Who is the author of the book \u0027To Kill a Mockingbird\u0027?",
+    "Answer": "Harper Lee",
+    "Topic": "Literature"
+}
+```
+
+## Running an evaluation
+
+Once a QA or a collection of QAs is available, the EvalLLM library can run the evaluation against the collection. The following image shows the console output of 2 different evaluated QAs.
+
+![2 hardcoded QAs evaluated](img/402QAEvalComplete.png)
+
+This the necessary code to run the eval of a QA:
+```csharp
+var qaProcessor = new QACreator.QACreator(kernelTest);
+var qa = new QA
+{
+    Question = "two plus two",
+    Answer = "'4' or 'four'",
+    Topic = "Math"
+};
+
+var processResult = await qaProcessor.Process(qa);
+var results = await batchEval.ProcessSingle(processResult);
+results.EvalRunName = "Harcoded QA 1";
+SpectreConsoleOutput.DisplayResults(results);
+```
+
+### Specifying the evaluate metrics
+
+The **LLMEval.Core** class will use the metrics specified while creating the object. These metrics are custom metrics that we've added based on the python sample: [Evaluating a RAG Chat App
+](https://github.com/Azure-Samples/ai-rag-chat-evaluator/tree/main).
+
+The following code shows how to add the 3 main custom evaluators:
+
+- coherence
+- groundedness
+- relevance
+- lenght
+
+```csharp
+// ========================================
+// create LLMEval and add evaluators
+// ========================================
+var kernelEvalFunctions = kernelEval.CreatePluginFromPromptDirectory("Prompts");
+var batchEval = new Core.LLMEval();
+
+batchEval
+    .AddEvaluator(new PromptScoreEval("coherence", kernelEval, kernelEvalFunctions["coherence"]))
+    .AddEvaluator(new PromptScoreEval("groundedness", kernelEval, kernelEvalFunctions["groundedness"]))
+    .AddEvaluator(new PromptScoreEval("relevance", kernelEval, kernelEvalFunctions["relevance"]))
+    .AddEvaluator(new LenghtEval());
+batchEval.SetMeterId("llama3");
+```
+
+#### Built-in metrics
```

Comment:
The distinction between built-in and custom metrics is a little confusing here. It sounds like Built-in is something that AI Studio provides built-in metrics and the ones we use here are the custom metrics. If that's the case, since custom metrics are the ones that are provided, I'd recommend moving those above built-in.

Reply:
I got the idea. I changed the approach to be:
1st explanation of the current implemented metrics
2nd small overview of Azure AI Studio Metrics, as reference